### PR TITLE
add custom artifact builder that patches in asthma remission

### PR DESCRIPTION
**Pull request by kfwilson**
this adds a custom artifact builder that will automatically patch in asthma remission data so it's easy to remake artifacts in one step

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,12 +18,13 @@
         'vivarium==0.8.21',
         'vivarium_public_health==0.9.13',
         'vivarium_cluster_tools==1.0.14',
-        'vivarium_inputs[data]==3.0.0',  # FIXME: artifact was built off dev version of inputs. this pin needs to be updated when released.
-        
+        'vivarium_inputs[data]==3.0.0',
+        # FIXME: artifact was built off dev version of inputs. this pin needs to be updated when released.
```
Comment (collijk) [inline setup.py]: Make a release of inputs if you need to.

File: src/vivarium_ihme_obesity_washout/tools/cli.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    model specification file stored in the repository.
+
+    It requires access to the J drive and /ihme. If you are running this job
+    from a qlogin on the cluster, you must specifically request J drive access
+    when you qlogin by adding "-l archive=TRUE" to your command.
+
+    Please have at least 30GB of memory on your qlogin."""
+    model_specification = Path(__file__).parent.parent / 'model_specifications' / 'vivarium_ihme_obesity_washout.yaml'
+    output_root = ARTIFACT_FOLDER / 'vivarium_ihme_obesity_washout'
+
```
Comment (collijk) [inline src/vivarium_ihme_obesity_washout/tools/cli.py:39]: We should think more about this pattern here.  Esp. w/r/t to locations.  A pattern that Rob introduced that I kind of like involves using template files with jinja.  E.g.:

Template file: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/external_data/yaml_template.in
function to generate concrete specs: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/external_data/build_simulation_files.py
CLI tool to generate model specs: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/cli.py#L30

Definitely not a thing we have to do now, but this approach requires hand editing the yaml files to create artifacts for multiple locs, which is a bit of a drag.  Especially if the underlying model changes (e.g. new causes/risks or alternative models for them).

File: src/vivarium_ihme_obesity_washout/tools/cli.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    utilities.setup_logging(output_root, verbose, None, model_specification, append)
+
+    try:
```
Comment (collijk) [inline src/vivarium_ihme_obesity_washout/tools/cli.py:42]: We should make a ticket to pull this pattern into the vivarium cli utilities.  I pulled it out here: https://github.com/ihmeuw/vivarium_development/blob/master/src/vadmin/utilities.py#L51 but we use it in almost every cli we build and vivarium is pretty much always a dependency.

File: src/vivarium_ihme_obesity_washout/tools/cli.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            traceback.print_exc()
+            pdb.post_mortem()
+        else:
+            raise
+
+
+def _patch_artifact(artifact_path: Path, model_specification: Path):
+    location = yaml.safe_load(model_specification.read_text())['configuration']['input_data']['location']
+    asthma = causes.asthma
+    asthma.dismod_id = ASTHMA_DISMOD_ID
```
Comment (collijk) [inline src/vivarium_ihme_obesity_washout/tools/cli.py:62]: This is also worth thinking about.  Maybe worthwhile to make a raw modelable entities mapping.

**Review by collijk — APPROVED** [review]
Only the release comment needs addressing at this time.

File: setup.py (hunk 1 above)
Comment (kfwilson) [reply setup.py]: release made and versions bumped in master branch

File: src/vivarium_ihme_obesity_washout/tools/cli.py (hunk 2 above)
Comment (halfhorst) [reply src/vivarium_ihme_obesity_washout/tools/cli.py:39]: this is interesting. I did something similar for the data requests but it was way more lightweight, just names in brackets and .format(). It worked well, the specs are autogenerated from a template with just different locations and file names

**Pull request merged**